pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1419,7 +1420,19 @@ public PinotResourceManagerResponse toggleTableState(String tableName, boolean s
       return new PinotResourceManagerResponse("Error: Table " + tableName + " not found.", false);
     }
     _helixAdmin.enableResource(_helixClusterName, tableName, status);
-    return (status) ? new PinotResourceManagerResponse("Table " + tableName + " successfully enabled.", true)
+
+    // If enabling a resource, also reset segments in error state for that resource
+    boolean resetSuccessful = false;
+    if (status) {
+      try {
+        _helixAdmin.resetResource(_helixClusterName, Collections.singletonList(tableName));
+        resetSuccessful = true;
+      } catch (HelixException e) {
+        LOGGER.warn("Caught exception while resetting resource {}, ignoring.", e, tableName);
```

Comment:
Recommend logging an error

Reply:
Warn is typically for things that did not work right but otherwise did not prevent the completion of the operation. Error is for things that completely prevented an operation.

In this case, the enabling went through, although it might be in a partial state.